src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,795 @@
+    using Castle.Components.DictionaryAdapter.Xml;
+    using DurableTask.AzureStorage.Storage;
+
+
+    /// <summary>
+    /// Partition ManagerV3 based on the Azure Storage V3.  
+    /// </summary>
+    sealed class TablePartitionManager : IPartitionManager, IDisposable
+    {
+        readonly AzureStorageClient azureStorageClient;
+        readonly AzureStorageOrchestrationService service;
+        readonly AzureStorageOrchestrationServiceSettings settings;
+        readonly CancellationTokenSource partitionManagerCancellationSource;
+        readonly string connectionString;
+        readonly string storageAccountName;
+        readonly TableClient partitionTable;
+        readonly TableLeaseManager tableLeaseManager;
+        readonly LeaseCollectionBalancerOptions options;
+
+        /// <summary>
+        /// constructor to initiate new instances of TablePartitionManager
+        /// </summary>
+        /// <param name="azureStorageClient">Client for the storage account.</param>
+        /// <param name="service">The service responsible for initiating or terminating the partition manager.</param>
+        public TablePartitionManager(
+            AzureStorageOrchestrationService service,
+            AzureStorageClient azureStorageClient)
+        {
+            this.azureStorageClient = azureStorageClient;
+            this.service = service;
+            this.settings = this.azureStorageClient.Settings;
+            this.connectionString = this.settings.StorageConnectionString ?? this.settings.StorageAccountDetails.ConnectionString;
+            this.storageAccountName = this.azureStorageClient.TableAccountName;
+            if(this.connectionString == null)
+            {
+                throw new Exception("Connection string is null. Managed identity is not supported in the table partition manager yet.");
+            }
+            this.options = new LeaseCollectionBalancerOptions
+            {
+                AcquireInterval = this.settings.LeaseAcquireInterval,
+                RenewInterval = this.settings.LeaseRenewInterval,
+                LeaseInterval = this.settings.LeaseInterval,
+                ShouldStealLeases = true
+            };
+            this.partitionManagerCancellationSource = new CancellationTokenSource();
+            this.partitionTable = new TableClient(this.connectionString, this.settings.PartitionTableName);
+            this.tableLeaseManager = new TableLeaseManager(this.partitionTable, this.service, this.settings, this.storageAccountName, this.options);
+        }
+
+        
+        /// <summary>
+        /// This method create a new instance of the class TableLeaseManager that represents the worker. 
+        /// And then start the loop that the worker keeps operating on the table. 
+        /// </summary>
+        async Task IPartitionManager.StartAsync()
+        {
+            await Task.Factory.StartNew(() => this.PartitionManagerLoop(this.partitionManagerCancellationSource.Token));
```

Comment:
I recommend this instead:
```suggestion
            _ = this.PartitionManagerLoop(this.partitionManagerCancellationSource.Token);
```

Reply:
Thanks! I have been struggling with this multi-threading method before. 

I used to implement like this, then the method should be like :
<img width="929" alt="image" src="https://github.com/Azure/durabletask/assets/110135109/5412a26f-d17f-4405-acbf-a61c887f61e6">

As you can see I need to remove `Async` cause we don't await anything here. But the method called `StartAsync` and the other two pm versions kindly use this `StartNew` method without any return. That's why I choose the current implementation. What do you think?

Reply:
You can't remove `Async` from the method name since that's part of the interface. However, you're not required to actually using async/await if you don't need it (in fact, you shouldn't since that introduces unnecessary overhead). I think your previous implementation was more correct. However, I would still prefer we further reduce the overhead and go with my suggestion above.